src/ui/hooks/useProfileHook.ts

```diff
@@ -48,181 +53,167 @@ export const useProfiles = () => {
+    return result;
+  }, []);
+
+  /**
+   * Sets up Ethereum Virtual Machine (EVM) wallet
+   * @param mainAddress The main wallet address to associate with EVM wallet
+   * @returns EVM wallet data object
+   * Called during initial profile setup and wallet creation
+   */
   const setupEvmWallet = useCallback(
-    async (mainAddress: string) => {
+    async (mainAddress: FlowAddress) => {
       try {
-        const [evmRes, emoji, evmWallet] = await Promise.all([
+        const [evmRes, emoji] = await Promise.all([
           usewallet.queryEvmAddress(mainAddress),
           usewallet.getEmoji(),
-          usewallet.getEvmWallet(),
         ]);
 
-        const evmAddress = ensureEvmAddressPrefix(evmRes!);
+        const evmAddress = evmRes ? ensureEvmAddressPrefix(evmRes) : '';
 
-        // Setup EVM wallet data
-        const evmWalletData: BlockchainResponse = {
-          ...evmWallet,
+        const evmWalletData: WalletAccount = {
           name: emoji[9].name,
           icon: emoji[9].emoji,
           address: evmAddress,
-          chain_id: '1',
+          chain: network === 'testnet' ? 545 : 747,
           id: 1,
-          coins: [],
           color: emoji[9].bgcolor,
         };
 
-        // Batch updates
-        await Promise.all([setEvmWallet(evmWalletData), setEvmAddress(evmAddress)]);
-        setEvmLoading(false);
-
-        return evmWalletData;
+        await Promise.all([setEvmWallet(evmWalletData), setEvmAddress(evmAddress ?? '')]);
       } catch (error) {
-        console.error('Error setting up EVM wallet:', error);
-        throw error;
+        console.error('Error processing EVM address:', error);
+      } finally {
+        setEvmLoading(false);
       }
     },
-    [usewallet, setEvmWallet, setEvmAddress, setEvmLoading]
+    [usewallet, network, setEvmWallet, setEvmAddress, setEvmLoading]
   );
 
-  // Helper function for fetchUserWallet
-  const freshUserInfo = useCallback(async () => {
-    if (!usewallet || !walletLoaded) return;
+  const fetchProfileData = useCallback(async () => {
+    if (profilesRef.current.loading || !usewallet || !walletLoaded) {
+      return;
+    }
+
     try {
-      // Make sure the wallet is unlocked and has a main wallet
-      if (!(await usewallet.getMainWallet())) {
-        console.log('freshUserInfo - No main wallet yet');
-        return;
-      }
-      //TODO: should rethink the wording of the wallet functions, have it be parent evm and child or something similar. State name and should be the same frontend and background.
-      const [currentWallet, isChild, mainAddress] = await Promise.all([
-        usewallet.getCurrentWallet(),
-        usewallet.getActiveWallet(),
-        usewallet.getMainAddress(),
-      ]);
-
-      if (!currentWallet) {
-        // We may not be logged in yet
-        return;
+      profilesRef.current.loading = true;
+      const mainAddress = await usewallet.getMainAddress();
+      console.log('mainAddress ===', mainAddress);
```

Comment:
this file still has quite a few debug logs, assuming they will be cleaned up before shipping